src/modules/position_estimator_inav/position_estimator_inav_main.cpp

```diff
@@ -215,27 +230,37 @@ static void write_debug_log(const char *msg, float dt, float x_est[2], float y_e
 	fclose(f);
 }
 #else
-#define write_debug_log(...) 
+#define write_debug_log(...)
 #endif
 
 /****************************************************************************
- * main
- ****************************************************************************/
+* main
+****************************************************************************/
 int position_estimator_inav_thread_main(int argc, char *argv[])
 {
 	int mavlink_fd;
 	mavlink_fd = px4_open(MAVLINK_LOG_DEVICE, 0);
 
-	float x_est[2] = { 0.0f, 0.0f };	// pos, vel
-	float y_est[2] = { 0.0f, 0.0f };	// pos, vel
-	float z_est[2] = { 0.0f, 0.0f };	// pos, vel
-
-	float est_buf[EST_BUF_SIZE][3][2];	// estimated position buffer
-	float R_buf[EST_BUF_SIZE][3][3];	// rotation matrix buffer
-	float R_gps[3][3];					// rotation matrix for GPS correction moment
-	memset(est_buf, 0, sizeof(est_buf));
-	memset(R_buf, 0, sizeof(R_buf));
-	memset(R_gps, 0, sizeof(R_gps));
+	InertialFilter inertial_filter;
+
+	Vector2f x_est = {0.0f, 0.0f};	// pos, vel
+	Vector2f y_est = {0.0f, 0.0f};	// pos, vel
+	Vector2f z_est = {0.0f, 0.0f};	// pos, vel
+
+	/* -*-* Eigen::Tensor usage *-*-
+	 * Tensor<float, 3, RowMajor> est_buf(EST_BUF_SIZE, 3, 2);	// estimated position buffer
+	 * est_buf.setZero();
+	 * Tensor<float, 3, RowMajor> R_buf(EST_BUF_SIZE, 3, 3);	// rotation matrix buffer
+	 * R_buf.setZero();
+	 */
+	std::vector<Matrix<float, 3, 2> > est_buf(EST_BUF_SIZE);// estimated position buffer
+	std::vector<Matrix3f> R_buf(EST_BUF_SIZE);	// rotation matrix buffer
+	for (size_t it = 0; it < EST_BUF_SIZE; it++) {
```

Comment:
Yeah I should use a `std::array` instead since the size is not dynamical and it's a lot more efficient. Didn't think of that. Will change then, thanks!